code/game/objects/structures/beds_chairs/chair.dm

```diff
@@ -107,10 +108,11 @@
 			buckled_mob.setDir(direction)
 
 /obj/structure/chair/proc/handle_layer()
-	if(has_buckled_mobs() && dir == NORTH)
-		layer = ABOVE_MOB_LAYER
-	else
-		layer = OBJ_LAYER
+  if(override_layer) return
```

Comment:
Dont use these style of inline conditionals, put the return on the new line.
Also override_Layer should be renamed to be more clear and should include a comment saying what it does, rather than just `// for benches`

Reply:
I've made the fixes but I believe that it should still be called override_layer as the variable that decides what displays above what is simply called layer everywhere else and all the variable does is override the chair layer code for benches